YoutubeExplode/Internal/YoutubeVideoStream.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace YoutubeExplode.Internal
+{
+    internal class YoutubeVideoStream : Stream
+    {
+        private readonly HttpClient _httpClient;
+
+        private const int MaxSegmentSize = 9_898_989;
+
+        private readonly string _url;
+
+        Stream _currentStream;
+
+        public YoutubeVideoStream(HttpClient httpClient, string url, long length)
+        {
+            _url = url;
+            _httpClient = httpClient;
+            Length = length;
+        }
+
+        public override bool CanRead => true;
+
+        public override bool CanSeek => true;
+
+        public override bool CanWrite => false;
+
+        public override long Length { get; }
+
+        public override long Position { get; set; }
+
+        public override void Flush() => throw new System.NotSupportedException();
+
+        public override async Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            if (Position >= Length)
+                return 0;
+
+            count = Math.Min(MaxSegmentSize - 1, count);
+            if(_currentStream == null)
+                _currentStream = await _httpClient.GetStreamAsync(_url, Position, Position + MaxSegmentSize).ConfigureAwait(false);
```

Comment:
I wonder could it be that `_url` might expire before everything is downloaded?

Reply:
I also noticed that before you would call `MaxSegmentSize - 1` I'm calling it here without `-1` and it seems to work ok, should I also add it or leave it as is right now?

Reply:
1) No, leave it as is
2) 0 -> SegmentSize - 1; SegmentSize -> SegmentSize+SegmentSize - 1; etc..
This was done to split into even portions. In your case it doesn't matter since you're not splitting into anything beforehand. That said, technically your segments are 1 byte bigger than MaxSegmentSize so I would still add -1.

Reply:
Added -1.